### PR TITLE
Add `pending_limit` into Scheduler (ref #19)

### PR DESCRIPTION
**Pull request by linw1995**
This PR for #19 
- Add parameter `pending_limit` into `Scheduler`
  for control pending jobs size
- Add property `pending_limit` into `Scheduler`
- Add property `waiting_count` into `Scheduler`

### CODE REVIEW

File: aiojobs/_scheduler.py [hunk 1]
```diff
@@ -10,20 +10,22 @@
     from collections.abc import Sized, Iterable, Container
     bases = Sized, Iterable, Container
 else:  # pragma: no cover
-    bases = (Collection,)
+    bases = (Collection, )
```
Comment (asvetlov) [inline aiojobs/_scheduler.py]: Please don't touch the line, space is not needed.

File: aiojobs/_scheduler.py [hunk 2]
```diff
@@ -71,8 +81,19 @@ def closed(self):
         if self._closed:
             raise RuntimeError("Scheduling a new job after closing")
         job = Job(coro, self, self._loop)
-        should_start = (self._limit is None or
-                        self.active_count < self._limit)
+        should_start = (self._limit is None
```
Comment (asvetlov) [inline aiojobs/_scheduler.py]: Please keep existing formatting for the line.
The same should be applied to `should_wait`

File: aiojobs/_scheduler.py [hunk 3]
```diff
@@ -60,6 +66,10 @@ def active_count(self):
     def pending_count(self):
         return len(self._pending)
 
+    @property
+    def waiting_count(self):
```
Comment (asvetlov) [inline aiojobs/_scheduler.py]: No need to expose the property IMHO

File: aiojobs/_scheduler.py [hunk 4]
```diff
@@ -71,8 +81,19 @@ def closed(self):
...
         job = Job(coro, self, self._loop)
-        should_start = (self._limit is None or
-                        self.active_count < self._limit)
+        should_start = (self._limit is None
+                        or self.active_count < self._limit)
+        should_wait = (self._pending_limit is not None
+                       and self.pending_count >= self._pending_limit)
+        if not should_start and should_wait:
+            waiter = self._loop.create_future()
+            waiter._job = job
```
Comment (asvetlov) [inline aiojobs/_scheduler.py]: No chance to accept this line: adding a custom attribute to asyncio Future has very bad smell.

File: aiojobs/_scheduler.py [hunk 5]
```diff
@@ -10,20 +10,22 @@
...
         self._loop = loop
         self._jobs = set()
         self._close_timeout = close_timeout
         self._limit = limit
+        self._pending_limit = pending_limit
         self._exception_handler = exception_handler
         self._failed_tasks = asyncio.Queue(loop=loop)
         self._failed_task = loop.create_task(self._wait_failed())
         self._pending = deque()
+        self._waiting = deque()
```
Comment (asvetlov) [inline aiojobs/_scheduler.py]: I believe using `asyncio.Queue` for `_pending` and not adding `_waiting` is more robust and simple solution.

Comment (linw1995) [reply aiojobs/_scheduler.py]: using `asyncio.Queue`  is simply, but can't limit the size of pending jobs , when set `pending_limit` == 0

Comment (asvetlov) [reply aiojobs/_scheduler.py]: `asyncio.Queue(maxsize=pending_limit)`, infinite size if `pending_limit=0`.
Do you see any problem with this solution?

Comment (linw1995) [reply aiojobs/_scheduler.py]: I mean `spawn` always waits until one active job closes when `pending_limit=0`. And  the `_pending` has infinite size when `pending_limit=None`

**Comment by asvetlov** [comment]
#24 is simpler

**Pull request closed without merging**